internal/host/host.go

```diff
@@ -585,9 +587,11 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 		_, err = hostutil.UpdateHostStatus(ctx, logutil.FromContext(ctx, m.log), m.db, m.eventsHandler, h.InfraEnvID, *h.ID,
 			swag.StringValue(h.Status), models.HostStatusError, statusInfo)
 	case models.HostStageRebooting:
-		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost {
+		if swag.StringValue(h.Kind) == models.HostKindAddToExistingClusterHost && !m.kubeApiEnabled{
```

Comment:
Maybe instead of passing another parameter you can check if the host have a kube-key? https://github.com/openshift/assisted-service/blob/7f0a41b671297e987ae07a6e55d32f82a5c0c974/internal/common/db.go#L78

Reply:
I think that `h` here is a models.Host, whereas the KubeKeyNamespace field is an attribute of common.Host. 
Not sure we can use this here?

Reply:
Make sense.